### ISSUE TITLE
Can't update to Meteor 1.3.5.1 - error in fourseven:scss package while updating

### ISSUE THREAD

**Issue by Twisterking** [fourseven/meteor-scss#194]
Hi there!

So I am on a Macbook developing my Meteor application since forever. Now I am running into problems updating to 1.3.5.1 (I think that's the newest version atm right?).
This project is currently running Meteor 1.3.4.1!

```
=> Errors while initializing project:

While loading package fourseven:scss@3.4.3:
error: Command failed:
module.js:340
throw err;
^
Error: Cannot find module '../'
at Function.Module._resolveFilename (module.js:338:15)
at Function.Module._load (module.js:280:25)
at Module.require (module.js:364:17)
at require (module.js:380:17)
at Object.<anonymous>
(/Users/Patrick/.meteor/packages/fourseven_scss/.3.4.3.10hu8xj++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/node_modules/.temp-zyo5c9/node_modules/node-sass/node_modules/cross-spawn/node_modules/spawn-sync/node_modules/try-thread-sleep/node_modules/thread-sleep/node_modules/.bin/node-pre-gyp:15:20)
at Module._compile (module.js:456:26)
at Object.Module._extensions..js (module.js:474:10)
at Module.load (module.js:356:32)
at Function.Module._load (module.js:312:12)
at Function.Module.runMain (module.js:497:10)

npm ERR! Darwin 15.5.0
npm ERR! argv "node"
"/Users/Patrick/.meteor/packages/meteor-tool/.1.3.5_1.8qvgc6++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/npm"
"rebuild" "--no-bin-links" "--update-binary"
npm ERR! node v0.10.46
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE
npm ERR! thread-sleep@1.0.4 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 8
npm ERR!
npm ERR! Failed at the thread-sleep@1.0.4 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the thread-sleep package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs thread-sleep
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls thread-sleep
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!
/Users/Patrick/.meteor/packages/fourseven_scss/.3.4.3.10hu8xj++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/node_modules/.temp-zyo5c9/npm-debug.log
```
Can someone help? Thanks a bunch in advance!

best, P

**Comment by sebakerckhof** [comment]
Could you try fourseven:scss@3.8.0

**Comment by Twisterking** [comment]
Thanks so much! It worked! :)
What I did:
run `meteor update fourseven:scss` first and normal `meteor update` afterwards.

**Comment by danieljvdm** [comment]
When I try to update to 3.8.0 I get the following:

```error: Conflict: Constraint caching-compiler@1.0.5_1 is not satisfied by caching-compiler 1.0.5.
Constraints on package "caching-compiler":
* caching-compiler@=1.0.5 <- top level
* caching-compiler@1.0.4 <- caching-html-compiler 1.0.6 <- static-html 1.0.8
* caching-compiler@1.0.5_1 <- fourseven:scss 3.8.0
* caching-compiler@1.0.5 <- coffeescript 1.1.0 <- cosmos:browserify 0.8.4 <- deanius:promise 3.1.3
* caching-compiler@1.0.5 <- coffeescript 1.1.0 <- practicalmeteor:mocha 2.4.5_2
* caching-compiler@1.0.0 <- cosmos:browserify 0.8.4 <- deanius:promise 3.1.3

Conflict: Constraint ecmascript@0.4.6_1 is not satisfied by ecmascript 0.4.4.
Constraints on package "ecmascript":
* ecmascript@=0.4.4 <- top level
* ecmascript@0.4.4 <- templating 1.1.10 <- launch-screen 1.0.12 <- mobile-experience 1.0.4
* ecmascript@0.4.4 <- templating 1.1.10 <- practicalmeteor:mocha 2.4.5_2
* ecmascript@0.4.3 <- random 1.0.9 <- accounts-base 1.2.7
* ecmascript@0.4.3 <- caching-html-compiler 1.0.6 <- static-html 1.0.8
* ecmascript@0.4.4 <- caching-compiler 1.0.5 <- caching-html-compiler 1.0.6 <- static-html 1.0.8
* ecmascript@0.4.4 <- caching-compiler 1.0.5 <- fourseven:scss 3.8.0
* ecmascript@0.4.3 <- templating-tools 1.0.4 <- caching-html-compiler 1.0.6 <- static-html 1.0.8
* ecmascript@0.4.3 <- templating-tools 1.0.4 <- static-html 1.0.8
* ecmascript@0.4.4 <- mongo 1.1.8
* ecmascript@0.4.4 <- allow-deny 1.0.5 <- mongo 1.1.8
* ecmascript@0.4.4 <- ddp-server 1.2.7 <- ddp 1.2.5 <- accounts-base 1.2.7
* ecmascript@0.4.4 <- webapp 1.2.9 <- autoupdate 1.2.10 <- hot-code-push 1.0.4 <- meteor-base 1.0.4
* ecmascript@0.4.4 <- webapp 1.2.9 <- autoupdate 1.2.10 <- practicalmeteor:mocha 2.4.5_2
* ecmascript@0.4.4 <- webapp 1.2.9 <- meteor-base 1.0.4
* ecmascript@0.4.3 <- webapp-hashing 1.0.9 <- webapp 1.2.9 <- autoupdate 1.2.10 <- hot-code-push 1.0.4 <- meteor-base 1.0.4
* ecmascript@0.4.3 <- webapp-hashing 1.0.9 <- webapp 1.2.9 <- autoupdate 1.2.10 <- practicalmeteor:mocha 2.4.5_2
* ecmascript@0.4.3 <- webapp-hashing 1.0.9 <- webapp 1.2.9 <- meteor-base 1.0.4
* ecmascript@0.4.4 <- reactive-dict 1.1.8 <- meteortoys:toykit 3.0.4 <- msavin:jetsetter 2.0.0 <- meteortoys:allthings 3.0.0
* ecmascript@0.4.4 <- reactive-dict 1.1.8 <- session 1.1.6
* ecmascript@0.4.4 <- http 1.1.6 <- autoupdate 1.2.10 <- hot-code-push 1.0.4 <- meteor-base 1.0.4
* ecmascript@0.4.4 <- http 1.1.6 <- autoupdate 1.2.10 <- practicalmeteor:mocha 2.4.5_2
* ecmascript@0.4.4 <- http 1.1.6 <- deanius:promise 3.1.3
* ecmascript@0.4.3 <- accounts-base 1.2.7
* ecmascript@0.4.4 <- accounts-password 1.1.10
* ecmascript@0.1.6 <- mdg:validation-error 0.5.1
* ecmascript@0.1.5 <- themeteorchef:bert 2.1.0
* ecmascript@0.4.6_1 <- fourseven:scss 3.8.0
* ecmascript@0.4.4 <- static-html 1.0.8
* ecmascript@0.4.1 <- xolvio:cleaner 0.3.1
* ecmascript@0.4.1 <- practicalmeteor:mocha 2.4.5_2
* ecmascript@0.4.4 <- coffeescript 1.1.0 <- cosmos:browserify 0.8.4 <- deanius:promise 3.1.3
* ecmascript@0.4.4 <- coffeescript 1.1.0 <- practicalmeteor:mocha 2.4.5_2
* ecmascript@0.4.0-beta.11 <- tmeasday:test-reporter-helpers 0.2.1 <- practicalmeteor:mocha 2.4.5_2
* ecmascript@0.4.1 <- xolvio:backdoor 0.2.0
* ecmascript@0.1.3 <- dburles:factory 1.0.0
* ecmascript@0.1.5 <- deanius:promise 3.1.3```

**Comment by sebakerckhof** [comment]
@danieljvdm which meteor version?

**Comment by bryanterrell** [comment]
I'm getting the same error as @danieljvdm - on METEOR@1.3.2.4.

**Comment by danieljvdm** [comment]
I'm on 1.3.3. Here's more error logs (I can't seem to get scss to target @3.8.0)

```
=> Errors while initializing project:         
                                              
While loading package fourseven:scss@3.3.3_3:
error: Command failed: 
module.js:340
throw err;
^
Error: Cannot find module '../'
at Function.Module._resolveFilename (module.js:338:15)
at Function.Module._load (module.js:280:25)
at Module.require (module.js:364:17)
at require (module.js:380:17)
at Object.<anonymous>
(/Users/daniel/.meteor/packages/fourseven_scss/.3.3.3_3.9p8fxt++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/compileScssBatch/node_modules/.temp-1lppjuf/node_modules/node-sass/node_modules/cross-spawn/node_modules/spawn-sync/node_modules/try-thread-sleep/node_modules/thread-sleep/node_modules/.bin/node-pre-gyp:15:20)
at Module._compile (module.js:456:26)
at Object.Module._extensions..js (module.js:474:10)
at Module.load (module.js:356:32)
at Function.Module._load (module.js:312:12)
at Function.Module.runMain (module.js:497:10)

npm ERR! Darwin 16.0.0
npm ERR! argv "node"
"/Users/daniel/.meteor/packages/meteor-tool/.1.3.5_1.1fgtulz++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/npm"
"rebuild" "--no-bin-links" "--update-binary"
npm ERR! node v0.10.46
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE
npm ERR! thread-sleep@1.0.4 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the thread-sleep@1.0.4 install script 'node-pre-gyp install
--fallback-to-build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the thread-sleep
package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs thread-sleep
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls thread-sleep
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!
/Users/daniel/.meteor/packages/fourseven_scss/.3.3.3_3.9p8fxt++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/compileScssBatch/node_modules/.temp-1lppjuf/npm-debug.log

module.js:340
throw err;
^
Error: Cannot find module '../'
at Function.Module._resolveFilename (module.js:338:15)
at Function.Module._load (module.js:280:25)
at Module.require (module.js:364:17)
at require (module.js:380:17)
at Object.<anonymous>
(/Users/daniel/.meteor/packages/fourseven_scss/.3.3.3_3.9p8fxt++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/compileScssBatch/node_modules/.temp-1lppjuf/node_modules/node-sass/node_modules/cross-spawn/node_modules/spawn-sync/node_modules/try-thread-sleep/node_modules/thread-sleep/node_modules/.bin/node-pre-gyp:15:20)
at Module._compile (module.js:456:26)
at Object.Module._extensions..js (module.js:474:10)
at Module.load (module.js:356:32)
at Function.Module._load (module.js:312:12)
at Function.Module.runMain (module.js:497:10)

npm ERR! Darwin 16.0.0
npm ERR! argv "node"
"/Users/daniel/.meteor/packages/meteor-tool/.1.3.5_1.1fgtulz++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/npm"
"rebuild" "--no-bin-links" "--update-binary"
npm ERR! node v0.10.46
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE
npm ERR! thread-sleep@1.0.4 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the thread-sleep@1.0.4 install script 'node-pre-gyp install
--fallback-to-build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the thread-sleep
package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs thread-sleep
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls thread-sleep
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!
/Users/daniel/.meteor/packages/fourseven_scss/.3.3.3_3.9p8fxt++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/node_modules/meteor/compileScssBatch/compileScssBatch/node_modules/.temp-1lppjuf/npm-debug.log
```

**Comment by allenfuller** [comment]
Running into the same issue. It seems something in Meteor 1.3.5.1 is trying to install fourseven:scss@3.3.3_3. Unable to remove or upgrade...

```
=> Errors while initializing project:         
                                              
While loading package fourseven:scss@3.3.3_3:
error: Command failed: 
module.js:340
throw err;
^
Error: Cannot find module '../'
at Function.Module._resolveFilename (module.js:338:15)
at Function.Module._load (module.js:280:25)
at Module.require (module.js:364:17)
at require (module.js:380:17)
at Object.<anonymous>
(/Users/allen/.meteor/packages/fourseven_scss/.3.3.3_3.1nwivqp++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/compileScssBatch/node_modules/.temp-bkur61/node_modules/node-sass/node_modules/cross-spawn/node_modules/spawn-sync/node_modules/try-thread-sleep/node_modules/thread-sleep/node_modules/.bin/node-pre-gyp:15:20)
at Module._compile (module.js:456:26)
at Object.Module._extensions..js (module.js:474:10)
at Module.load (module.js:356:32)
at Function.Module._load (module.js:312:12)
at Function.Module.runMain (module.js:497:10)

npm ERR! Darwin 15.6.0
npm ERR! argv "node" "/Users/allen/.meteor/packages/meteor-tool/.1.3.5_1.8grdll++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/npm" "rebuild" "--no-bin-links" "--update-binary"
npm ERR! node v0.10.46
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE
npm ERR! thread-sleep@1.0.4 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the thread-sleep@1.0.4 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the thread-sleep package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs thread-sleep
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls thread-sleep
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/allen/.meteor/packages/fourseven_scss/.3.3.3_3.1nwivqp++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/compileScssBatch/node_modules/.temp-bkur61/npm-debug.log

module.js:340
throw err;
^
Error: Cannot find module '../'
at Function.Module._resolveFilename (module.js:338:15)
at Function.Module._load (module.js:280:25)
at Module.require (module.js:364:17)
at require (module.js:380:17)
at Object.<anonymous>
(/Users/allen/.meteor/packages/fourseven_scss/.3.3.3_3.1nwivqp++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/compileScssBatch/node_modules/.temp-bkur61/node_modules/node-sass/node_modules/cross-spawn/node_modules/spawn-sync/node_modules/try-thread-sleep/node_modules/thread-sleep/node_modules/.bin/node-pre-gyp:15:20)
at Module._compile (module.js:456:26)
at Object.Module._extensions..js (module.js:474:10)
at Module.load (module.js:356:32)
at Function.Module._load (module.js:312:12)
at Function.Module.runMain (module.js:497:10)

npm ERR! Darwin 15.6.0
npm ERR! argv "node" "/Users/allen/.meteor/packages/meteor-tool/.1.3.5_1.8grdll++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/npm" "rebuild" "--no-bin-links" "--update-binary"
npm ERR! node v0.10.46
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE
npm ERR! thread-sleep@1.0.4 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the thread-sleep@1.0.4 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the thread-sleep package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs thread-sleep
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls thread-sleep
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/allen/.meteor/packages/fourseven_scss/.3.3.3_3.1nwivqp++os.osx.x86_64+web.browser+web.cordova/plugin.compileScssBatch.os.osx.x86_64/npm/compileScssBatch/node_modules/.temp-bkur61/npm-debug.log
```

**Comment by allenfuller** [comment]
Finally got the app to upgrade. Here's the steps I followed:

1. Remove `~/.meteor` (the home meteor app) by running `sudo rm -rf ~/.meteor`
2. Remove the project's /.meteor/local folder
3. Remove the project's /node_modules folder
4. Reinstall Meteor by running `curl https://install.meteor.com/ | sh`
5. Update the project's /.meteor/release file to "METEOR@1.4.0.1"
6. Remove any reference to fourseven:scss from project/.meteor/packages and project/.meteor/versions
7. Run `meteor add fourseven:scss@3.8.0_1`
8. Run `meteor` in the project directory. It will download the meteor-tool for the current version of meteor

Whew! I actually ended up removing my local repository entirely and cloning from remote, but I don't think that's necessary after going through these steps.

Hope this helps!
